apis/core.oam.dev/v1alpha1/applyonce_types.go

```diff
@@ -16,12 +16,51 @@ limitations under the License.
 
 package v1alpha1
 
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
 const (
 	// ApplyOncePolicyType refers to the type of configuration drift policy
 	ApplyOncePolicyType = "apply-once"
 )
 
 // ApplyOncePolicySpec defines the spec of preventing configuration drift
 type ApplyOncePolicySpec struct {
-	Enable bool `json:"enable"`
+	Enable bool                  `json:"enable"`
+	Rules  []ApplyOncePolicyRule `json:"rules"`
+}
+
+// ApplyOncePolicyRule defines a single apply-once policy rule
+type ApplyOncePolicyRule struct {
+	Selector ResourcePolicyRuleSelector `json:"selector"`
+	Strategy ApplyOnceStrategy          `json:"strategy"`
+}
+
+// ApplyOnceStrategy the strategy for resource path to allow configuration drift
+type ApplyOnceStrategy struct {
+	// Path the specified path that allow configuration drift
+	// like 'spec.template.spec.containers[0].resources' and '*' means the whole target allow configuration drift
+	Path []string `json:"path"`
+}
+
+// FindStrategy find apply-once strategy for target resource
+func (in ApplyOncePolicySpec) FindStrategy(manifest *unstructured.Unstructured) *ApplyOnceStrategy {
```

Comment:
```suggestion
func (in *ApplyOncePolicySpec) FindStrategy(manifest *unstructured.Unstructured) *ApplyOnceStrategy {
```

Reply:
FindStrategy => ExtractStrategy 

Will that be more clear?

Reply:
FindStrategy keep consistent with GarbageCollectPolicySpec.FindStrategy